File: src/FamilyGroup/Family.java
package FamilyGroup;

import java.util.ArrayList;
import java.util.Scanner;


public class Family <U extends MemberFamily>  implements AddFamilyInterface {

    public ArrayList<U> addMan(ArrayList<U> allFamily) {

        Male father = new Male();

        Scanner sc = new Scanner(System.in);
        System.out.println("Введите имя: ");
        String name = sc.next();
        System.out.println("Введите фамилию: ");
        String secondName = sc.next();
        System.out.println("Введите имя отца: ");
        String nameFather = sc.next();
        System.out.println("Введите имя матери: ");
        String nameMother = sc.next();
        System.out.println("Введите возраст: ");
        int age = sc.nextInt();
        System.out.println("Введите имя жены: ");
        String nameWife = sc.next();

        father.setFirstName(name);
        father.setSecondName(secondName);
        father.setNameFather(nameFather);
        father.setNameMother(nameMother);
        father.setAge(age);
        father.setNameWife(nameWife);

        allFamily.add((U) father);

        return allFamily;
    }

    public ArrayList<U> addWoman(ArrayList<U> allFamily) {

        Female mother = new Female();

        Scanner sc = new Scanner(System.in);
        System.out.println("Введите имя: ");
        String name = sc.next();
        System.out.println("Введите фамилию: ");
        String secondName = sc.next();
        System.out.println("Введите имя отца: ");
        String nameFather = sc.next();
        System.out.println("Введите имя матери: ");
        String nameMother = sc.next();
        System.out.println("Введите возраст: ");
        int age = sc.nextInt();
        System.out.println("Введите имя мужа: ");
        String nameHusband = sc.next();

        mother.setFirstName(name);
        mother.setSecondName(secondName);
        mother.setNameFather(nameFather);
        mother.setNameMother(nameMother);
        mother.setAge(age);
        mother.setNameHusband(nameHusband);

        allFamily.add((U) mother);

        return allFamily;

    }

}

File: PythonOOP/FamilyPack/Family.py
from Creator import Creator
from FamilyPack import Male, Female


class Family(Creator):

    def create_man(self):
        first_name = input('Введите имя: ')
        second_name = input('Введите фамилию: ')
        name_father = input('Введите имя отца: ')
        name_mother = input('Введите имя матери: ')
        age = input('Введите возраст: ')
        name_wife = input('Введите имя жены:')
        man = Male(first_name, second_name, name_father, name_mother, age, name_wife)
        return man

    def create_woman(self):
        first_name = input('Введите имя: ')
        second_name = input('Введите фамилию: ')
        name_father = input('Введите имя отца: ')
        name_mother = input('Введите имя матери: ')
        age = input('Введите возраст: ')
        name_husband = input('Введите имя мужа:')
        woman = Female(first_name, second_name, name_father, name_mother, age, name_husband)
        return woman


